Sources/Subprocess/Configuration.swift

```diff
@@ -815,34 +594,83 @@ internal struct TrackedFileDescriptor {
         self.closeWhenDone = closeWhenDone
     }
 
-    internal func safelyClose() throws {
+    #if os(Windows)
+    consuming func consumeDiskIO() -> FileDescriptor {
+        let result = self.fileDescriptor
+        // Transfer the ownership out and therefor
+        // don't perform close on deinit
+        self.closeWhenDone = false
+        return result
+    }
+    #endif
+
+    internal mutating func safelyClose() throws {
         guard self.closeWhenDone else {
             return
         }
+        closeWhenDone = false
 
         do {
             try fileDescriptor.close()
         } catch {
             guard let errno: Errno = error as? Errno else {
                 throw error
             }
-            if errno != .badFileDescriptor {
-                throw errno
+            // Getting `.badFileDescriptor` suggests that the file descriptor
+            // might have been closed unexpectedly. This can pose security risks
+            // if another part of the code inadvertently reuses the same file descriptor
+            // number. This problem is especially concerning on Unix systems due to POSIX’s
+            // guarantee of using the lowest available file descriptor number, making reuse
+            // more probable. We use `fatalError` upon receiving `.badFileDescriptor`
+            // to prevent accidentally closing a different file descriptor.
+            guard errno != .badFileDescriptor else {
+                fatalError(
+                    "FileDescriptor \(fileDescriptor.rawValue) is already closed"
+                )
             }
+            // Throw other kinds of errors to allow user to catch them
+            throw error
         }
     }
 
-    internal var platformDescriptor: PlatformFileDescriptor {
+    deinit {
+        guard self.closeWhenDone else {
+            return
+        }
+
+        do {
+            try fileDescriptor.close()
+        } catch {
+            guard let errno: Errno = error as? Errno else {
+                return
+            }
+            // Getting `.badFileDescriptor` suggests that the file descriptor
+            // might have been closed unexpectedly. This can pose security risks
+            // if another part of the code inadvertently reuses the same file descriptor
+            // number. This problem is especially concerning on Unix systems due to POSIX’s
+            // guarantee of using the lowest available file descriptor number, making reuse
+            // more probable. We use `fatalError` upon receiving `.badFileDescriptor`
+            // to prevent accidentally closing a different file descriptor.
+            guard errno != .badFileDescriptor else {
+                fatalError(
+                    "FileDescriptor \(fileDescriptor.rawValue) is already closed"
+                )
+            }
+            // Otherwise ignore the error
```

Comment:
this is a bad idea. Closing in `deinit` for file descriptors is bad enough (`close` may block too depending on the fd) but ignoring the errors isn't a thing we can do.

Reply:
Yup, agree with Franz. `with*` solves those issues correctly and also clearly for a reader of the code.

Reply:
@weissi can you open up issues for your comments? This PR is merged and discussions here might be lost.